### ISSUE TITLE
[FEATURE]: Upgrade Gazebo from Garden to Harmonic

### ISSUE THREAD

**Issue by evan-palmer** [Robotic-Decision-Making-Lab/blue#170]
### Feature Type

Changing existing functionality in the BlueROV2 driver

### Problem Description

The current LTS (Jazzy) and Rolling use Gazebo Harmonic instead of Gazebo Garden.

### Feature Description

Upgrade the `main` and `jazzy` branches to use this version of Gazebo.

### Alternative Solutions

N/A

**Comment by evan-palmer** [comment]
This is currently being blocked by https://github.com/mavlink/mavros/issues/1958. Once merged, we can resolve this issue.

**Comment by evan-palmer** [comment]
Harmonic is now the recommended version for Humble and Iron. Might be good to upgrade those first while waiting for Jazzy support

**Comment by amarburg** [comment]
#220 upgrades Rolling to Harmonic.  I could break just the upgrade to Harmonic into a separate PR which you could propagate to Humble and Iron (leaving the Rolling / Noble - specific changes in the existing PR)

**Comment by evan-palmer** [comment]
I just double-checked this, and [I was wrong](https://gazebosim.org/docs/harmonic/ros_installation/). Hypothetically, the backport should still work (`ros_gz` is already being built from source to support Garden), but I'm contemplating discontinuing the `ros_gz` source build in future releases to reduce some of the maintenance and build times (and also make things more stable). If we move to using the `ros_gz` binaries in #220, then we will probably want to keep separate PRs.

**Comment by amarburg** [comment]
So here's a merge of the ROS-Gazebo compatibility matrix and the ROS-Gazebo-ros_gz matrix:

| ROS2 Version | Gazebo Version | ROS2-Gazebo Compat | ros_gz binaries? |
|-----|------|-----|----|
| Humble | Fortress | **Recommended** | [https://packages.ros.org/](https://packages.ros.org/) |
| Humble | Garden | Possible | [gazebo packages](https://gazebosim.org/docs/latest/ros_installation#gazebo-garden-with-ros-2-humble-iron-or-rolling-use-with-caution-) |
| Humble | Harmonic | Possible | [gazebo packages](https://gazebosim.org/docs/latest/ros_installation#gazebo-garden-with-ros-2-humble-iron-or-rolling-use-with-caution-) |
| Iron | Fortress | **Recommended** | [https://packages.ros.org/](https://packages.ros.org/)|
| Iron | Garden | Possible | From source |
| Iron | Harmonic | Possible | From source |
| Jazzy | Garden | Possible | From source |
| Jazzy | Harmonic | **Recommended** | [https://packages.ros.org/](https://packages.ros.org/)|
| Rolling | Fortress | **Recommended** | [https://packages.ros.org/](https://packages.ros.org/)|
| Rolling | Garden | Possible | From source |
| Rolling | Harmonic | Possible | From source |

Based on the spotty availability of binaries, what's the desirable endpoint for blue?  

Here's where we are now:

| ROS2 Version | Gazebo Version | ros_gz |
|---------------|----------------|--------|
| Humble | Garden | From source |
| Iron | Garden | From source |
| Jazzy | Harmonic ([PR work-in-progress here](https://github.com/apl-ocean-engineering/blue/pull/1))| From source (but packages are available) |
| Rolling | Garden (or Harmonic via PR #220) | From source (packages available for Harmonic) |

I assume Rolling is converging with Jazzy (i.e. Harmonic will soon be the preferred version)?

**Comment by evan-palmer** [comment]
Here is what I'm thinking:

| ROS 2 Version | Gazebo Version | ros_gz binaries |
| :-------------------: | :---------------------: | :--------------------: |
| Humble  | Harmonic | [packages.osrfoundation.org](http://packages.osrfoundation.org/) |
| Iron | Harmonic | [packages.osrfoundation.org](http://packages.osrfoundation.org/) |
| Jazzy | Harmonic | [packages.ros.org](packages.ros.org) |
| Rolling | Harmonic | Open to discuss |

The reason that I am undecided on Rolling is because it may be nice to switch to new versions of Gazebo as they come out, but I am open to discuss that. [Here](https://gazebosim.org/docs/harmonic/ros_installation/#gazebo-harmonic-with-ros-2-humble-iron-or-rolling-use-with-caution) are the docs on the Iron/Humble/Jazzy Harmonic installation.

**Comment by amarburg** [comment]
Hm.   So I'm stuck on how to implement the switching between ros_gz sources from a git-mergify point of view.   How about:

1. This PR switches the default to harmonic.   Apply it to `main` (aka rolling) and propagate to `jazzy` `iron` and `humble`.  
2. Targeted commit to `jazzy` which removes the source build for `ros_gz` and instead installs from packages.ros.org
3. Targeted commit to `iron` which removes the source build and instead installs from `osrffoundation.org`
